packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -3035,48 +3042,49 @@ class StackViewport extends Viewport {
     viewRef: ViewReference,
     options: ReferenceCompatibleOptions = {}
   ): boolean {
-    if (!super.isReferenceViewable(viewRef, options)) {
-      return false;
-    }
-
-    const { referencedImageId, sliceIndex } = viewRef;
-
-    if (viewRef.volumeId && !referencedImageId) {
-      return options.asVolume;
-    }
-
-    let testIndex = this.getCurrentImageIdIndex();
-    let currentImageId = this.imageIds[testIndex];
-
-    if (options.withNavigation && typeof sliceIndex === 'number') {
-      testIndex = sliceIndex;
-      currentImageId = this.imageIds[testIndex];
-    }
-
-    if (!currentImageId) {
+    const testIndex = this.getCurrentImageIdIndex();
+    const currentImageId = this.imageIds[testIndex];
+    if (!currentImageId || !viewRef) {
```

Comment:
How there is a case where there is no `currentImageId`? I don't see how `this.imageIds[this.getCurrentImageIdIndex]` become undefined

Reply:
The viewport is uninitialized is primarily what I was concerned with, but if you are ok with it throwing, then we could ignore it.

Reply:
it's fine leave it